flang/lib/Optimizer/Dialect/FIROps.cpp

```diff
@@ -36,6 +37,18 @@ namespace {
 #include "flang/Optimizer/Dialect/CanonicalizationPatterns.inc"
 } // namespace
 
+static void propagateAttributes(mlir::Operation *fromOp,
+                                mlir::Operation *toOp) {
+  if (!fromOp || !toOp)
+    return;
+
+  for (mlir::NamedAttribute attr : fromOp->getAttrs()) {
+    if (attr.getName().str().rfind(
```

Comment:
nit: can we use `starts_with` here to make it more readable?

Reply:
I might have overlooked it but std::string does not have `start_with`?